pkg/webhook/cluster.go

```diff
@@ -0,0 +1,67 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+type ClusterWebhook struct {
+}
+
+func (r *ClusterWebhook) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		WithValidator(r).
+		For(&v1beta1.Cluster{}).
+		Complete()
+}
+
+var _ webhook.CustomValidator = &ClusterWebhook{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *ClusterWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	cluster, ok := obj.(*v1beta1.Cluster)
+	if !ok {
+		panic("expected to get an of object of type v1beta1.Cluster")
+	}
+	switch cluster.Spec.InfrastructureRef.Kind {
+	case "AWSCluster":
+	case "AzureCluster":
+	case "GCPCluster":
```

Comment:
Nit, can inline these
```suggestion
	case "AWSCluster", "AzureCluster", "GCPCluster":
```